File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/DeliveryAddress.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

import java.util.Objects;

/**
 * Address where goods can be delivered to.
 */
public class DeliveryAddress {

    private String firstName;
    private String lastName;
    private String companyName;
    private Address address;

    /**
     * @return firstName
     */
    public String getFirstName() {
        return firstName;
    }
    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }
    public DeliveryAddress withFirstName(String firstName) {
        this.firstName = firstName;
        return this;
    }

    /**
     * @return lastName
     */
    public String getLastName() {
        return lastName;
    }
    public void setLastName(String lastName) {
        this.lastName = lastName;
    }
    public DeliveryAddress withLastName(String lastName) {
        this.lastName = lastName;
        return this;
    }

    /**
     * @return companyName
     */
    public String getCompanyName() {
        return companyName;
    }
    public void setCompanyName(String companyName) {
        this.companyName = companyName;
    }
    public DeliveryAddress withCompanyName(String companyName) {
        this.companyName = companyName;
        return this;
    }

    /**
     * @return address
     */
    public Address getAddress() {
        return address;
    }
    public void setAddress(Address address) {
        this.address = address;
    }
    public DeliveryAddress withAddress(Address address) {
        this.address = address;
        return this;
    }

    @Override
    public boolean equals(java.lang.Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        DeliveryAddress deliveryAddress = (DeliveryAddress) o;
        return Objects.equals(this.firstName, deliveryAddress.firstName)
                && Objects.equals(this.lastName, deliveryAddress.lastName)
                && Objects.equals(this.companyName, deliveryAddress.companyName)
                && Objects.equals(this.address, deliveryAddress.address);
    }

    @Override
    public int hashCode() {
        return Objects.hash(firstName, lastName, companyName, address);
    }

    @Override
    public String toString() {
        return "class DeliveryAddress {\n"
                + "        firstName: " + toIndentedString(firstName) + "\n"
                + "        lastName: " + toIndentedString(lastName) + "\n"
                + "        companyName: " + toIndentedString(companyName) + "\n"
                + "        address: " + toIndentedString(address) + "\n"
                + "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
     */
    private String toIndentedString(java.lang.Object o) {
        return o == null
                ? "null"
                : o.toString().replace("\n", "\n        ");
    }
}

File: upl-purchase-sdk-js/src/unzer-paylater/model/DeliveryAddress.js
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

const ModelHelper = require("../util/ModelHelper");
const Address = require("./Address");

/**
 * Address where goods can be delivered to.
 */
class DeliveryAddress {
    /**
     * @returns { String }
     */
    getFirstName() {
        return this.firstName;
    }
    /**
     * @param { String } firstName
     */
    setFirstName(firstName) {
        this.firstName = ModelHelper.validatePrimitive(firstName, "string");
    }
    /**
     * @param { String } val
     */
    withFirstName(val) {
        this.setFirstName(val);
        return this;
    }

    /**
     * @returns { String }
     */
    getLastName() {
        return this.lastName;
    }
    /**
     * @param { String } lastName
     */
    setLastName(lastName) {
        this.lastName = ModelHelper.validatePrimitive(lastName, "string");
    }
    /**
     * @param { String } val
     */
    withLastName(val) {
        this.setLastName(val);
        return this;
    }

    /**
     * @returns { String }
     */
    getCompanyName() {
        return this.companyName;
    }
    /**
     * @param { String } companyName
     */
    setCompanyName(companyName) {
        this.companyName = ModelHelper.validatePrimitive(companyName, "string");
    }
    /**
     * @param { String } val
     */
    withCompanyName(val) {
        this.setCompanyName(val);
        return this;
    }

    /**
     * @returns { Address }
     */
    getAddress() {
        return this.address;
    }
    /**
     * @param { Address } address
     */
    setAddress(address) {
        this.address = ModelHelper.validateObject(address, Address);
    }
    /**
     * @param { Address } val
     */
    withAddress(val) {
        this.setAddress(val);
        return this;
    }

    /**
     * @returns { DeliveryAddress }
     */
    static constructFromObject(data) {
        if (!data) return undefined;
        return new DeliveryAddress()
            .withFirstName(ModelHelper.convertToType(data["firstName"], String))
            .withLastName(ModelHelper.convertToType(data["lastName"], String))
            .withCompanyName(ModelHelper.convertToType(data["companyName"], String))
            .withAddress(ModelHelper.convertToType(data["address"], Address));
    }
}

module.exports = DeliveryAddress;
